core/arch/arm/kernel/thread.c

```diff
@@ -370,9 +370,9 @@ static void init_regs(struct thread_ctx *thread,
 	thread->regs.pc = (uint64_t)thread_std_smc_entry;
 
 	/*
-	 * Stdcalls starts in SVC mode with masked IRQ, masked Asynchronous
-	 * abort and unmasked FIQ.
-	  */
+	 * Stdcalls starts in SVC mode with masked foreign interrupts, masked
+	 * Asynchronous abort and unmasked native interrupts.
+	 */
 	thread->regs.cpsr = SPSR_64(SPSR_64_MODE_EL1, SPSR_64_MODE_SP_EL0,
 				    DAIFBIT_IRQ | DAIFBIT_ABT);
```

Comment:
This sets `DAIFBIT_IRQ`. It should set the foreign interrutps, depending on GIC version.

Could it be the same in AArch32 `init_regs()` above (with `CPSR_I`)?